analysis/staticAnalysisReport.Rmd

```diff
@@ -0,0 +1,106 @@
+---
+title: "Eval Static Analysis"
+author: "Chakshu Goyal"
+date: "07/06/2018"
+output:
+  html_document: default
+  pdf_document: default
+---
+
+## 
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(ggplot2)
+library(tidyverse)
+library(tools)
+```
+
+## Load the data
+Usage of the function eval in R-packages
+```{r}
+evaldf <- read_csv("evals.csv")
+```
+
+## Plot occurences of evals in packages
+```{r one}
+# Occurences of evals in each package plot
+ggplot(evaldf , aes(pkg)) + geom_bar() + theme(axis.text.x = element_blank()) + labs(y="Number of evals", x="packages", title="Figurative distribution of the usage of eval in various packages")
+
+# No. of occurences of eval in each package table
+evalCntPerPkg <- plyr::count(evaldf , "pkg")
+knitr::kable(head(arrange(evalCntPerPkg, desc(freq)) , 20) , col.names = c("Package" , "Number of evals"))
+
+#plot histogram for number of projects vs number of evals
+ggplot(evalCntPerPkg, aes(freq)) + geom_density()+ scale_x_log10() + labs(y="Number of projects (density)", x="Number of Evals - (log scaled)", title="Evals distribution in the packages")
+ggplot(evalCntPerPkg , aes(freq)) + geom_histogram(bins=50) + scale_x_log10() + labs(y="Number of projects", x="Number of Evals", title="Evals distribution in the packages - histogram")
+ggplot(evalCntPerPkg , aes(x = "" , y = freq)) + geom_boxplot() + labs(y="Number of Evals", x="Packages", title="Evals distribution in the packages - boxplot")
+
+##outliers
+outlyingVals <- boxplot(evalCntPerPkg$freq, plot = F)$out
+outlyingRows <- filter(evalCntPerPkg, freq %in% outlyingVals)
+##top outliers
+knitr::kable(head(arrange(outlyingRows, desc(freq)) , 5) , col.names = c("Package" , "Number of evals"))
+##bottom outliers
+knitr::kable(head(arrange(outlyingRows, freq) , 5) , col.names = c("Package" , "Number of evals"))
+
+#plotting without the outliers
+evalCntPerPkgMod <- evalCntPerPkg[ !(evalCntPerPkg$freq %in% outlyingVals), ]
```

Comment:
It might be easier to write:
```r
evalCntPerPkg %>%
  filter(!(freq %in% outlyingVals)) %>%
  ggplot(aes(x = "" , y = freq)) + 
  geom_boxplot() + 
  labs(y="Number of Evals", x="Packages", title="Evals distribution in the packages - boxplot (outliers removed)")
```

Reply:
Yes, thanks!